radicle-node/src/service.rs

```diff
@@ -443,16 +468,17 @@ where
                 };
                 debug!(
                     target: "service",
-                    "Found {} connected seed(s) and {} unconnected seed(s) for {}",
-                    connected.len(), unconnected.len(), rid
+                    "Found {} connected seed(s), {} disconnected seed(s), and {} fetching seed(s) for {}",
+                    stats.connected, stats.disconnected, stats.fetching, rid
                 );
-                resp.send(connected).ok();
+                resp.send(seeds).ok();
             }
             Command::Fetch(rid, seed, resp) => {
                 // TODO: Establish connections to unconnected seeds, and retry.
                 // TODO: Fetch requests should be queued and re-checked to see if they can
                 //       be fulfilled everytime a new node connects.
                 self.fetch_reqs.insert(rid, resp);
+                self.fetching_from.insert(seed);
```

Comment:
You can do without this field entirely by checking the session state I think. Just check that the protocol is `Protocol::Fetch`. Perhaps you could add a `Session::is_fetching` method to simplify things.

Reply:
Ah, thanks! That makes more sense :grin: